File: FrontEnd/aurofin-frontend/src/app/entity/InstallmentDto.ts
import { InstallmentStatus } from './installment-status.enum.js';

export interface InstallmentDto {
  installmentId: string; // UUID
  instAmt: number;
  instStartDate: string; // ISO date string
  instEndDate: string; // ISO date string
  installmentStatus: InstallmentStatus;
  paidDate?: string; // ISO date string
  fineAmt?: number;
}

File: Backend/src/main/java/com/lendingApp/main/dto/InstallmentDto.java
package com.lendingApp.main.dto;

import java.time.LocalDate;
import java.util.UUID;

import com.lendingApp.main.enums.InstallmentStatus;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
@Data
@NoArgsConstructor
@AllArgsConstructor
public class InstallmentDto {
	private UUID installmentId;
    private Double instAmt;
    private LocalDate instStartDate;
    private LocalDate instEndDate;
    private InstallmentStatus installmentStatus;
    private LocalDate paidDate;
    private Double fineAmt;
}
